deeplearning4j-cli/deeplearning4j-cli-api/src/main/java/org/deeplearning4j/cli/subcommands/Train.java

```diff
@@ -59,294 +61,229 @@
+                LayerFactory factory = LayerFactories.getFactory(conf);
+                Layer l = factory.create(conf);
+                DataSetIterator iter = new RecordReaderDataSetIterator( reader , conf.getBatchSize());
+                while(iter.hasNext()) {
+                    l.fit(iter.next().getFeatureMatrix());
+                }
 
-	}
+                if(saveMode.equals("binary")) {
+                    BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(this.outputDirectory));
+                    DataOutputStream dos = new DataOutputStream(bos);
+                    Nd4j.write(l.params(),dos);
+                }
+                else {
+                    Nd4j.writeTxt(l.params(),outputDirectory,",");
+                }
 
-	public void execOnHadoop() {
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+
+    }
+
+    public void execOnSpark() {
+        log.warn( "DL4J: Execution on spark from CLI not yet supported" );
+    }
+
+    public void execOnHadoop() {
+        log.warn( "DL4J: Execution on hadoop from CLI not yet supported" );
+    }
+
+    /**
+     * Create an input format
+     * @return the input format to be created
+     */
+    public InputFormat createInputFormat() {
 
-		log.warn( "DL4J: Execution on hadoop from CLI not yet supported" );
+        //log.warn( "> Loading Input Format: " + (String) this.configProps.get( INPUT_FORMAT ) );
 
-	}
+        String clazz = (String) this.configProps.get( INPUT_FORMAT_KEY );
 
-	public InputFormat createInputFormat() {
+        if ( null == clazz ) {
+            clazz = DEFAULT_INPUT_FORMAT_CLASSNAME;
+        }
 
-		//log.warn( "> Loading Input Format: " + (String) this.configProps.get( INPUT_FORMAT ) );
+        try {
+            Class<? extends InputFormat> inputFormatClazz = (Class<? extends InputFormat>) Class.forName(clazz);
+            return inputFormatClazz.newInstance();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
 
-		String clazz = (String) this.configProps.get( INPUT_FORMAT_KEY );
+    }
 
-		if ( null == clazz ) {
-			clazz = DEFAULT_INPUT_FORMAT_CLASSNAME;
-		}
 
-		try {
-			Class<? extends InputFormat> inputFormatClazz = (Class<? extends InputFormat>) Class.forName(clazz);
-			return inputFormatClazz.newInstance();
-		} catch (Exception e) {
-			throw new RuntimeException(e);
-		}
+    public void loadConfigFile() throws Exception {
+
+        this.configProps = new Properties();
+
+        InputStream in = null;
+        try {
+            in = new FileInputStream( this.configurationFile );
+        } catch (FileNotFoundException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        try {
+            this.configProps.load(in);
+            in.close();
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+
```

Comment:
Replace this entire try {} by 
{Code}

try (InputStream in = new FileInputStream(this.configurationFile)) {
  this.configProps.load(in);
}